### PR TITLE
full --filter support with dates using jouch

### LINKED ISSUE #179: filter by complex filter strings
rather than `tick log -t projectA dev -d "mar 1-31"` we could filter with `tick log -f "#projectA and #dev mar 1-31"`

### PR DESCRIPTION
**Pull request by jonotron**
Breaking change

This is a fairly significant change as it now uses [pouch-find](https://github.com/nolanlawson/pouchdb-find) and [jouch](https://github.com/tickbin/jouch) to convert expressions like "#tagA or #tagB Jan - Feb" to find entries that match.

Fixes #179

### CODE REVIEW

**Comment by jonotron** [comment]
Fixes #179 

Currently this is a WIP, but I'm placing it here for others to peek at. Will try to resolve outstanding problems tomorrow.

**Comment by jonotron** [comment]
@tickbin/core This PR introduces significant changes, I wouldn't mind another set of eyes at some point before merging this in.

I've introduced pouch-find as the query library, which would change how upgrades query the database but I did not provide an implementation to handle upgrades with these new indexes. Should I include such a change in this PR? or issue another before release?

Since this does add a drastic change to the querying should we create a candidate branch for this to merge other fixes into? then merge into master once stable?

**Comment by chrisfosterelli** [comment]
@jonotron Will the previous upgrade scripts simply not work? I'm confused about the new `/bin/reindex` file, does that transition one user from the previous indices to find-indices? Why wouldn't we just make that as an upgrade script?

**Comment by jonotron** [comment]
I think the previous upgrader will work for now. `/bin/reindex` is more-or-less a util for development... it removes previous indexes and then re-indexes. When I was developing this I would occasionally need to nuke them and there isn't an easy way to do it, so I created that utility. I will document that more clearly.

**Comment by chrisfosterelli** [comment]
Seems reasonable, I think it'd be good to merge this in then :+1:

**Comment by jonotron** [comment]
Just realized some documentation needs to be included:

* [ ] Update README with usage documentation
* [ ] Update command examples with usage

File: src/query.js [hunk 1]
```diff
@@ -23,25 +24,32 @@ export default class Query {
...
+    const filterDates = dates[0] ? dates[0].text : defaultDatesFilter 
+    const reconstructed = parsed ? `(${parsed}) and ${filterDates}` : filterDates
+
+    const selector = jouch(reconstructed)
+    this._find = {
+      selector,
+      sort: ['startArr'] 
+    }
+
+    this._chain = this._chain//.filter(filter)
```
Comment (chadfawcett) [inline src/query.js]: is `.filter(filter)` supposed to be commented out here?

Comment (jonotron) [inline src/query.js]: Yes, though it should really be removed entirely, not just commented out.

Comment (jonotron) [inline src/query.js]: fixed in #203

**Pull request merged**